cmd/openshift-install/create.go

```diff
@@ -128,3 +144,47 @@ func runTargetCmd(targets ...asset.WritableAsset) func(cmd *cobra.Command, args
 		return nil
 	}
 }
+
+// FIXME: pulling the kubeconfig and metadata out of the root
+// directory is a but cludgy when we already have them in memory.
+func destroyBootstrap(ctx context.Context, directory string) (err error) {
+	kubeconfig, err := ioutil.ReadFile(filepath.Join(directory, "auth", "kubeconfig"))
+	if err != nil {
+		return err
+	}
+
+	config, err := clientcmd.BuildConfigFromFlags("", string(kubeconfig))
+	if err != nil {
+		return err
+	}
+
+	client, err := corev1.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	watcher, err := client.Events("kube-system").Watch(metav1.ListOptions{Watch: true})  // FIXME: FieldSelector?  LabelSelector?
+	if err != nil {
+		return err
+	}
+
+	logrus.Info("wait for bootstrap completion")
+	event, err := watchtools.UntilWithoutRetry(
```

Comment:
from the godocs
```
Warning: Unless you have a very specific use case (probably a special Watcher) don't use this function!!! Warning: This will fail e.g. on API timeouts and/or 'too old resource version' error. Warning: You are most probably looking for a function Until or UntilWithSync below, Warning: solving such issues. TODO: Consider making this function private to prevent misuse when the other occurrences in our codebase are gone.
```

This func will fail when api is not reachable, that happens a lot in bootstrapping

https://godoc.org/k8s.io/apimachinery/pkg/util/wait#PollImmediate might be enough?

Reply:
This call is from [here][1] and uses the `watch.go` robustifiers copied from that PR.  But I'll copy over the `Until` func too to make that more clear.

[1]: https://github.com/kubernetes/kubernetes/pull/50102/files#diff-b7ca7f71d0c92e184f946e29d5734a4eR104

Reply:
> But I'll copy over the `Until` func too to make that more clear.

Done with 223f200 -> c898075.